tests/shared/src/test/scala/cats/tests/HashMapSuite.scala

```diff
@@ -0,0 +1,292 @@
+  )
+  checkAll("UnorderedTraverse[HashMap[Int, *]]",
+           SerializableTests.serializable(HashMap.catsDataUnorderedTraverseForHashMap[Int])
+  )
+
+  checkAll("HashMap[Int, String]", CommutativeMonoidTests[HashMap[Int, Int]].commutativeMonoid)
+  checkAll("CommutativeMonoid[HashMap[Int, Int]]",
+           SerializableTests.serializable(HashMap.catsDataCommutativeMonoidForHashMap[Int, Int])
+  )
+
+  // Key-value pairs with the last binding for each distinct key
+  def distinctBindings[K, V](kvs: List[(K, V)]) =
+    kvs.reverse.distinctBy(_._1).reverse
+
+  test("show") {
+    assert(HashMap("a" -> 1, "b" -> 2, "c" -> 3).show === "HashMap(c -> 3, a -> 1, b -> 2)")
+    assert(HashMap.empty[String, Int].show === "HashMap()")
+  }
+
+  test("isEmpty and nonEmpty") {
+    assert(HashMap.empty[Int, Int].isEmpty)
+    forAll { (kvs: List[(Int, String)]) =>
+      val hashMap = HashMap.fromSeq(kvs)
+      assert(hashMap.isEmpty === kvs.isEmpty)
+      assert(hashMap.nonEmpty === kvs.nonEmpty)
+    }
+  }
+
+  test("===") {
+    forAll { (kvs: List[(Int, String)]) =>
+      val left = HashMap.fromSeq(kvs)
+      val right = HashMap.fromSeq(kvs)
+      assert(left === right)
+    }
+
+    forAll { (leftKvs: List[(Int, String)], rightKvs: List[(Int, String)]) =>
+      val left = HashMap.fromSeq(leftKvs)
+      val right = HashMap.fromSeq(rightKvs)
+      assert((distinctBindings(leftKvs) === distinctBindings(rightKvs)) === (left === right))
+    }
+  }
+
+  test("size") {
+    assert(HashMap.empty[Int, String].size === 0)
+    assert(HashMap(1 -> "a", 2 -> "b", 3 -> "c").size === 3)
+    assert(HashMap("Aa" -> 1, "BB" -> 2).size == 2)
+
+    forAll { (kvs: List[(Int, String)]) =>
+      val hashMap = HashMap.fromSeq(kvs)
+      assert(hashMap.size === distinctBindings(kvs).size)
+    }
+  }
+
+  test("get") {
+    forAll { (kvs: List[(Int, String)]) =>
+      val hashMap = HashMap.fromSeq(kvs)
+      distinctBindings(kvs).foreach { case (k, v) =>
+        hashMap.get(k) === Some(v)
+      }
+    }
+  }
+
+  test("concat") {
+    forAll { (leftKvs: List[(Int, String)], rightKvs: List[(Int, String)]) =>
+      val distinctKvs = distinctBindings(leftKvs ++ rightKvs)
+      val left = HashMap.fromSeq(leftKvs)
+      val right = HashMap.fromSeq(rightKvs)
+      val both = left.concat(right)
+      assert(both === HashMap.fromSeq(leftKvs ++ rightKvs))
+      distinctKvs.foreach { case (k, v) =>
+        assert(both.contains(k))
+        assert(both.get(k) === Some(v))
+      }
+    }
+  }
+
+  property("Empty HashMap never contains") {
+    forAll { (i: Int) =>
+      assert(!HashMap.empty[Int, String].contains(i))
+    }
+  }
+
+  property("Empty HashMap add consistent with contains") {
+    forAll { (i: Int, s: String) =>
+      assert(HashMap.empty[Int, String].add(i, s).contains(i))
+    }
+  }
+
+  property("Empty HashMap add and remove consistent with contains") {
```

Comment:
I think these tests would be better by generating a random HashMap from a closed set of operations:

```scala
def genHashMap[K, V](genK: Gen[K], genV: Gen[V])(implicit hashK: Hash[K]): Gen[HashMap[K, V]] =
  ...
  either:
  HashMap.empty,
  HashMap.fromFoldable,
  recurse.updated(k, v)
  recurse.remove(k)
```
So, basically randomly generate a hashmap by randomly doing all of the base operations we expose.